src/qml/qgismobileapp.qml

```diff
@@ -2006,6 +2006,11 @@ ApplicationWindow {
               geometryRequested = false
             }
           }
+
+          if(homeButton.waitingForDigitizingFinish){
```

Comment:
I think you need to turn that variable to false when users also successfully complete a digitizing action ( i.e  in onConfirmed )

For the `homeButton.waitingForDigitizingFinish = false` below, I'd do that in the openWelcomeScreen() function, that way you can de-duplicate the call that's also in the homeButton onClicked function.

Also, formatting: 
```suggestion
          if (homeButton.waitingForDigitizingFinish) {
```

Reply:
Appreciate your review! Correct, i'll fix them :)